src/GitVersion.Core/VersionCalculation/TrunkBasedVersionCalculator.cs

```diff
@@ -0,0 +1,7 @@
+namespace GitVersion.VersionCalculation;
+
+internal sealed class TrunkBasedVersionCalculator : ITrunkBasedVersionCalculator
+{
+    // TODO: Please implement trunk based version here and remove MainlineVersionCalculator.
+    public SemanticVersion Calculate(NextVersion nextVersion) => throw new NotImplementedException();
```

Comment:
In which circumstance may this exception be thrown? Is it possible to exercise the code in a way that this exception is invoked? Either way I would like the exception to have a user-friendly message.

```suggestion
    public SemanticVersion Calculate(NextVersion nextVersion) =>
        throw new NotImplementedException("Trunk based version calculation is not yet implemented. Use Mainline mode instead.");
```

Reply:
This code will never be called but it is a reminder to go ahead with the development. I have placed a TODO description as well.

Reply:
Done